File: app/src/main/java/com/kickstarter/mock/factories/InternalBuildEnvelopeFactory.java
package com.kickstarter.mock.factories;

import com.kickstarter.services.apiresponses.InternalBuildEnvelope;

import androidx.annotation.NonNull;

public final class InternalBuildEnvelopeFactory {
  private InternalBuildEnvelopeFactory() {}

  public static @NonNull InternalBuildEnvelope internalBuildEnvelope() {
    return InternalBuildEnvelope.builder()
      .build(123456)
      .changelog("Bug fixes")
      .newerBuildAvailable(false)
      .build();
  }

  public static @NonNull InternalBuildEnvelope newerBuildAvailable() {
    return internalBuildEnvelope().toBuilder()
      .newerBuildAvailable(true)
      .build();
  }
}

File: app/src/main/java/com/kickstarter/mock/factories/InternalBuildEnvelopeFactory.kt
package com.kickstarter.mock.factories

import com.kickstarter.services.apiresponses.InternalBuildEnvelope

object InternalBuildEnvelopeFactory {
    fun internalBuildEnvelope(): InternalBuildEnvelope {
        return InternalBuildEnvelope.builder()
            .build(123456)
            .changelog("Bug fixes")
            .newerBuildAvailable(false)
            .build()
    }

    @JvmStatic
    fun newerBuildAvailable(): InternalBuildEnvelope {
        return internalBuildEnvelope().toBuilder()
            .newerBuildAvailable(true)
            .build()
    }
}
